network-api/networkapi/wagtailpages/customblocks.py

```diff
@@ -41,6 +41,20 @@ class Meta:
         template = 'wagtailpages/blocks/image_block.html'
 
 
+class AirTableBlock(blocks.StructBlock):
```

Comment:
Should this be `class AirTableBlock(iFrameBlock):`?

Reply:
Possibly- what's the difference?

Reply:
Exactly: we already have an iframe block for embedding an iframe with some arbitrary source into a page, so if this injects an iframe then most of this class code is just duplication and we don't need to write it.

Reply:
one thing I just remember, we'll need to add airtable to our CSP config (however we're doing that) if we're loading it in an iframe

Reply:
yeah, that's part of what the iframe block makes clear: its helptext informs users that whatever domain they add, it _must_ get added to the domain whitelist.

There's an issue open about seeing if we can make start making that a CMS-managable (or really, enrichable) list over on https://github.com/mozilla/foundation.mozilla.org/issues/2342

Reply:
After looking at the iFrameBlock model and template, I'd strongly argue that this is different enough that it shouldn't inherit. The use case for this is more of a single item on a page. It also embeds a script on the page specifically for AirTable iframes, with some custom attributes that Airtable embeds give us. There's nothing of the existing iFrameBlock that I want to reuse, including the help text. If a user is selecting an element called "Airtable", they shouldn't be reminded that it needs to be approved; that's on us to do the one time we land this code. (With that in mind, I've added a commit to add Airtable to the CSP defaults)

Reply:
It feels like it might be worth inventing some kind of iframe factory since we now have iframe, youtube, and airtable, and they all "Sort of do the same, with different parameters", but that can be a story for a different year.

Reply:
I can see why you'd say that. But given the troubles we've had in the past, would the extra layer of abstraction actually yield any benefits or would it just get in our way when it comes to enhancing/changing the models in the future?